weave/integrations/openai/openai_sdk.py

```diff
@@ -0,0 +1,286 @@
+                            tool_call["function"]["arguments"] += (
+                                tool_call_delta.function.arguments
+                            )
+
+        return acc_choices
+
+    if acc is None:
+        if hasattr(value, "choices"):
+            output_choices = _process_chunk(value)
+            acc = ChatCompletionChunk(
+                id=value.id,  # Each chunk has the same ID
+                choices=output_choices,
+                created=value.created,  # Each chunk has the same timestamp
+                model=value.model,
+                object=value.object,
+                system_fingerprint=value.system_fingerprint,
+            )
+            return acc
+        else:
+            raise ValueError("Initial event must contain choices")
+
+    output_choices = _process_chunk(
+        value, [choice.model_dump() for choice in acc.choices]
+    )
+
+    acc = ChatCompletionChunk(
+        id=acc.id,
+        choices=output_choices,
+        created=acc.created,
+        model=acc.model,
+        object=acc.object,
+        system_fingerprint=acc.system_fingerprint,
+    )
+
+    # add usage info
+    if len(value.choices) == 0 and value.usage:
+        acc.usage = value.usage
+        if skip_last:
+            raise StopIteration(acc)
+
+    return acc
+
+
+# Unlike other integrations, streaming is based on input flag
+def should_use_accumulator(inputs: typing.Dict) -> bool:
+    return isinstance(inputs, dict) and bool(inputs.get("stream"))
+
+
+def create_wrapper(name: str) -> typing.Callable[[typing.Callable], typing.Callable]:
+    def wrapper(fn: typing.Callable) -> typing.Callable:
+        "We need to do this so we can check if `stream` is used"
+
+        def _add_stream_options(fn: typing.Callable) -> typing.Callable:
+            @wraps(fn)
+            def _wrapper(*args: typing.Any, **kwargs: typing.Any) -> typing.Any:
+                if bool(kwargs.get("stream")) and kwargs.get("stream_options") is None:
+                    kwargs["stream_options"] = {"include_usage": True}
+                return fn(
+                    *args, **kwargs
+                )  # This is where the final execution of fn is happening.
+
+            return _wrapper
+
+        def _openai_stream_options_is_set(inputs: typing.Dict) -> bool:
+            if inputs.get("stream_options") is not None:
+                return True
+            return False
+
+        op = weave.op()(_add_stream_options(fn))
+        op.name = name  # type: ignore
+        return add_accumulator(
+            op,  # type: ignore
+            make_accumulator=lambda inputs: lambda acc, value: openai_accumulator(
+                acc, value, skip_last=not _openai_stream_options_is_set(inputs)
+            ),
+            should_accumulate=should_use_accumulator,
+            on_finish_post_processor=openai_on_finish_post_processor,
+        )
+
+    return wrapper
+
+
+symbol_pathers = [
```

Comment:
@ayulockin minor nit: typo here -> should be `symbol_patchers` instead of `symbol_pathers`